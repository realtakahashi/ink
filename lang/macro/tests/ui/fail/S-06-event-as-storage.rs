use ink_lang as ink;

#[ink::contract(version = "0.1.0")]
mod event_as_storage {
    #[ink(event)]
    #[ink(storage)] // We cannot have #[ink(storage)] if we already have #[ink(event)]
    struct EventAsStorage {}

    impl EventAsStorage {
        #[ink(constructor)]
        fn constructor() -> Self {
            Self {}
        }

        #[ink(message)]
        fn message(&self) {}
    }
}

fn main() {}
