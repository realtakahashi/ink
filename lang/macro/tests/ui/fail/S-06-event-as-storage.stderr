error: invalid ink! attribute found for `#[ink(event)]` struct
 --> $DIR/S-06-event-as-storage.rs:6:10
  |
6 |     #[ink(storage)] // We cannot have #[ink(storage)] if we already have #[ink(event)]
  |          ^^^^^^^^^
